### ISSUE TITLE
Match

### ISSUE THREAD

**Issue by Ayc0** [Ayc0/react-only#4]
```js
<Match>
  <Component1 only="md"> // Only display this component on md
    ...
  </Component1>
  <Component2 only="xlDown" />  // Only display this component on xlDown
  ...
</Match>
```

Inspired by Reach Router (see [createRoute](https://github.com/reach/router/blob/master/src/index.js#L474) and [Router](https://github.com/reach/router/blob/master/src/index.js#L167))

Behavior: Parse props of children and displays them if their inner media query matches the webview

Should we parse inner props too?

**Comment by Ayc0** [comment]
Possible implementation:

```js
const parseChildren = cb => (element) => {
  const children = (element.props && element.props.children) ?
    React.Children.map(children, parseChildren(cb)) :
    null;
  const { only, ...props } = element.props || {};
  const clone = React.cloneElement(element, props, children);
  if (!only) {
    return clone;
  }
  return <Only on={only}>{clone}</Only>
}
```

**Comment by Ayc0** [comment]
https://codepen.io/ayc0/pen/ejPGwx?editors=0010

**Comment by Ayc0** [comment]
included in c61e553581abaf9b2933d59a18d6208648652dec
